pkg/kubecost/allocationfilter.go

```diff
@@ -80,6 +80,19 @@ type AllocationFilter interface {
 	// Matches is the canonical in-Go function for determing if an Allocation
 	// matches a filter.
 	Matches(a *Allocation) bool
+
+	// Flattened converts a filter into a minimal form, removing unnecessary
+	// intermediate objects, like single-element or zero-element AND and OR
+	// conditions.
+	//
+	// It returns nil if the filter is filtering nothing.
+	//
+	// Example:
+	// (and (or (namespaceequals "kubecost")) (or)) ->
+	// (namespaceequals "kubecost")
+	//
+	// (and (or)) -> nil
+	Flattened() AllocationFilter
```

Comment:
If this returns a new `AllocationFilter` can we call the function `Flatten()`?

Reply:
I picked `Flattened()` from Python influence -- `[].sort()` modifies in place and `sorted([])` returns a new list. I'm happy to use either name; does go have any idiomatic naming for functions like this?

Reply:
I don't know that there is a specific convention in Go around verb tense. If you like `Flattened`, I won't argue! In general, we've tended to use present-tense, but it really doesn't matter too much.